src/services/readwise-api.ts

```
@@ -67,17 +67,32 @@ export default class ReadwiseApi {
 
+      if (!response.ok && response.status !== 429) {
+        console.error(`Readwise: Failed to fetch data. Status: ${response.status} - ${response.statusText}`);
+        throw new Error(`Failed to fetch data. Status: ${response.status}`);
+      }
+
       if (response.status === 429) {
         // Error handling for rate limit throttling
-        const rateLimitedDelayTime = parseInt(response.headers.get('Retry-After')) * 1000 + 1000;
-        console.warn(`Readwise: API Rate Limited, waiting to retry for ${rateLimitedDelayTime}`);
+        let rateLimitedDelayTime = parseInt(response.headers.get('Retry-After')) * 1000 + 1000;
+        if (isNaN(rateLimitedDelayTime)) {
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)